### PR TITLE
#1236 Added Organization Users unit test

### PR DESCRIPTION
**Pull request by hritikmunde**
## Type of change

- [x] I have added tests for `addOrganizationUser`
- [x] I have added tests for `getOrganizationUsers`
- [x] I have added tests for `getOrganizationUser`
- [x] I have added tests for `getOrganizationUsersCount`
- [x] I have added tests for `deleteOrganizationUser`

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox

### CODE REVIEW

**Comment by kevkevinpal** [comment]
```
Failed to compile.

TS2304: Cannot find name 'mockFetch'.
    85 |     const mockApiResponse = { status: 200, message: 'success' };
    86 |
  > 87 |     mockFetch.mockReturnValueOnce(Promise.resolve(mockApiResponse));
       |     ^^^^^^^^^
    88 |
    89 |     const organizationUser = {
    90 |       owner_pubkey: user.owner_pubkey || '',


error Command failed with exit code 1.
```

**Comment by ecurrencyhodler** [comment]
@hritikmunde you have a couple of tests failing. Can you fix those? We like to have all tests pass before reviewing. Thanks!

**Comment by hritikmunde** [comment]
@ecurrencyhodler , made changes please try again and let me know. Thanks!

**Comment by hritikmunde** [comment]
@ecurrencyhodler my apologies, just ran prettier. Pushing changes

**Comment by hritikmunde** [comment]
@ecurrencyhodler , can you check if any of the tests I added are failing?

**Comment by hritikmunde** [comment]
@ecurrencyhodler , Please run again, Thanks!

**Comment by hritikmunde** [comment]
@ecurrencyhodler , any update on the review of this?

**Comment by ecurrencyhodler** [comment]
Looks like Jest is still failing. Can you try again?

**Comment by hritikmunde** [comment]
The tests failing are in main.ts and I can’t see my tests failed there but still I can try again. On my local though all the tests are passed.

**Comment by ecurrencyhodler** [comment]
@elraphty or @kevkevinpal do you have any suggestions here?

**Comment by hritikmunde** [comment]
@ecurrencyhodler , is there any way for me to push changes and run the workflow to test. I'll be working today afternoon which will be a different time zone for you, mostly night.

**Comment by ecurrencyhodler** [comment]
@hritikmunde I'm not sure. I should be around later in the evening. Just ping me on sphinx chat and I'll run it manually.

**Comment by hritikmunde** [comment]
Okay sure, thanks!

**Comment by hritikmunde** [comment]
Hi @ecurrencyhodler , please find below the tests failing. As you can see the tests I added have passed.
image
Please confirm who has added the failed tests and how we can get them running again. Also, this file conflicted with issue #1234 .

**Comment by hritikmunde** [comment]
@kevkevinpal , Any suggestion on this? I guess the tests for bounties have been tested in the last workflow and the hardcoded data was used up leading to the deletion of the mock bounty on the last test in issue #1234. Hence the tests are failing. I can see my test cases are working fine.

**Comment by wright-eric** [comment]
> @kevkevinpal , Any suggestion on this? I guess the tests for bounties have been tested in the last workflow and the hardcoded data was used up leading to the deletion of the mock bounty on the last test in issue #1234. Hence the tests are failing. I can see my test cases are working fine.

Yup i have used mocked data for mocked api calls. I think the issue here maybe using mockFetch instead of `sinon`.

**Comment by hritikmunde** [comment]
@ecurrencyhodler , please specify what to be done.

**Comment by hritikmunde** [comment]
> > @kevkevinpal , Any suggestion on this? I guess the tests for bounties have been tested in the last workflow and the hardcoded data was used up leading to the deletion of the mock bounty on the last test in issue #1234. Hence the tests are failing. I can see my test cases are working fine.
> 
> 
> 
> Yup i have used mocked data for mocked api calls. I think the issue here maybe using mockFetch instead of `sinon`.

MockFetch shouldn't hinder with sinon. I used mock data too. As you can see the Expected and received results are different mostly in boolean values.

**Comment by kevkevinpal** [comment]
@hritikmunde I think what @wright-eric said is correct if you can use sinon instead in this file that would probably work best as I think mock fetch and sinon are conflicting since they are both stubbing `global.fetch`

**Comment by hritikmunde** [comment]
Let me try that. Will update here once done

**Comment by hritikmunde** [comment]
Hi @kevkevinpal , Here are the tests with Sinon. Still, those tests are failing. There is a possibility now that its the issue with tests, not the mockFetch.
image

**Comment by kevkevinpal** [comment]
> Hi @kevkevinpal , Here are the tests with Sinon. Still, those tests are failing. There is a possibility now that its the issue with tests, not the mockFetch. image @wright-eric maybe you can check the mock data you used.

what is the error you are getting? if you've modified `./src/config/host.ts` you need to revert cause some of the tests will fail

**Comment by kevkevinpal** [comment]
This is my local on master seems to be working
Screenshot 2024-01-05 at 12 18 22 PM

you need to modify your host.ts because some of the test use that file to run tests. so if locally it is changed then they will fail. You will not see host.ts modified on your git status because it is in the .gitignore

**Comment by hritikmunde** [comment]
> This is my local on master seems to be working Screenshot 2024-01-05 at 12 18 22 PM
> 
> you need to modify your host.ts because some of the test use that file to run tests. so if locally it is changed then they will fail. You will not see host.ts modified on your git status because it is in the .gitignore

Ofc I'm testing by reverting the host. Maybe I can push the changes done now and we can test again. Is that okay? Now that I have added sinon maybe it'll work this time.

**Comment by hritikmunde** [comment]
@ecurrencyhodler , can you run the workflow again?

**Comment by kevkevinpal** [comment]
looks like the tests pass now

**Comment by ecurrencyhodler** [comment]
All checks passed!

**Comment by hritikmunde** [comment]
> looks like the tests pass now

Yepp thanks for the help. Hell of a test!

File: frontend/app/src/store/__test__/main.spec.ts [hunk 1]
```diff
@@ -10,14 +10,22 @@ import mockBounties, { expectedBountyResponses } from '../../bounties/__mock__/m
...
 afterAll(() => {
-  jest.clearAllMocks();
+  global.fetch = origFetch;
+
+  sinon.restore();
 });
 
+const getOrganizationUsersEndpoint = (orgUUID: string): string =>
+  `${TribesURL}/organizations/users/${orgUUID}`;
+
```
Comment (kevkevinpal) [inline frontend/app/src/store/__test__/main.spec.ts:28]: I dont see the point of defining this for a single use, we can just use the string directly in the test

File: frontend/app/src/store/__test__/main.spec.ts [hunk 2]
```diff
@@ -75,6 +83,180 @@ describe('Main store', () => {
...
+    fetchStub.resolves(Promise.resolve(mockApiResponse));
+
+    const orgUUID = 'cmas9gatu2rvqiev4ur0';
+
+    const endpoint = getOrganizationUsersEndpoint(orgUUID);
+
+    const users = await mainStore.getOrganizationUsers(orgUUID);
+
+    sinon.assert.calledWithMatch(fetchStub, endpoint, sinon.match.any);
+    expect(users).toEqual([{ uuid: 'cm3eulatu2rvqi9o75ug' }, { uuid: 'cldl1g04nncmf23du7kg' }]);
```
[the remark this answers is not in the document]
Comment (hritikmunde) [reply frontend/app/src/store/__test__/main.spec.ts:136]: Sure I'll make the changes

**Comment by kevkevinpal** [comment]
added a few comments but overall looks good @elraphty can you give a quick review aswell?

File: frontend/app/src/store/__test__/main.spec.ts [hunk 3]
```diff
@@ -75,6 +83,180 @@ describe('Main store', () => {
...
+    const mockApiResponse = {
+      status: 200,
+      json: sinon.stub().resolves({
+        uuid: 'cm3eulatu2rvqi9o75ug'
+      })
+    };
+
+    fetchStub.resolves(Promise.resolve(mockApiResponse));
+
+    const userUUID = 'cm3eulatu2rvqi9o75ug';
```
[the remark this answers is not in the document]
Comment (hritikmunde) [reply frontend/app/src/store/__test__/main.spec.ts:151]: Okay got it

File: frontend/app/src/store/__test__/main.spec.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hritikmunde) [reply frontend/app/src/store/__test__/main.spec.ts:136]: You mean uuid?

**Pull request merged**